File: app/src/main/java/al/ahgitdevelopment/municion/Licencia.java
package al.ahgitdevelopment.municion;

import java.util.Date;

/**
 * Created by Alberto on 13/05/2016.
 */
public class Licencia {
    private int id;
    private String tipo;
    private int num_licencia;
    private Date fechaExpedicion;
    private Date fechaCaducidad;

    public int getId() {
        return id;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public int getNum_licencia() {
        return num_licencia;
    }

    public void setNum_licencia(int num_licencia) {
        this.num_licencia = num_licencia;
    }

    public Date getFechaExpedicion() {
        return fechaExpedicion;
    }

    public void setFechaExpedicion(Date fechaExpedicion) {
        this.fechaExpedicion = fechaExpedicion;
    }

    public Date getFechaCaducidad() {
        return fechaCaducidad;
    }

    public void setFechaCaducidad(Date fechaCaducidad) {
        this.fechaCaducidad = fechaCaducidad;
    }
}

File: app/src/main/java/al/ahgitdevelopment/municion/datamodel/Licencia.kt
package al.ahgitdevelopment.municion.datamodel

import al.ahgitdevelopment.municion.repository.dao.*
import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable

/**
 * Created by Alberto on 13/05/2016.
 */
@Entity(tableName = TABLE_LICENCIAS)
data class Licencia(

        @PrimaryKey(autoGenerate = true)
        @NonNull
        @ColumnInfo(name = KEY_ID) var id: Long,

        @ColumnInfo(name = KEY_LICENCIAS_TIPO) var tipo: Int,

        @ColumnInfo(name = KEY_LICENCIAS_NOMBRE) var nombre: String,

        @ColumnInfo(name = KEY_LICENCIAS_TIPO_PERMISO_CONDUCCION) var tipoPermisoConduccion: Int,

        @ColumnInfo(name = KEY_LICENCIAS_EDAD) var edad: Int,

        @ColumnInfo(name = KEY_LICENCIAS_FECHA_EXPEDICION) var fechaExpedicion: String,

        @ColumnInfo(name = KEY_LICENCIAS_FECHA_CADUCIDAD) var fechaCaducidad: String,

        @ColumnInfo(name = KEY_LICENCIAS_NUM_LICENCIA) var numLicencia: String,

        @ColumnInfo(name = KEY_LICENCIAS_NUM_ABONADO) var numAbonado: Int,

        @ColumnInfo(name = KEY_LICENCIAS_NUM_SEGURO) var numSeguro: String,

        @ColumnInfo(name = KEY_LICENCIAS_AUTONOMIA) var autonomia: Int,

        @ColumnInfo(name = KEY_LICENCIAS_ESCALA) var escala: Int,

        @ColumnInfo(name = KEY_LICENCIAS_CATEGORIA) var categoria: Int

) : Serializable
